### ISSUE TITLE
[clang-tidy] readability-qualified-auto and std::array iterators

### ISSUE THREAD

**Issue by ecorm** [llvm/llvm-project#63461]
`readability-qualified-auto` suggests `const auto*` when initializing an `auto` variable with a std::array iterator value that happens to be implemented as a raw pointer.

The suggested change is not portable, as the standard only requires that a `contiguous_iterator` be returned, which does not necessarily have to be a raw pointer.

Example:
```c++
#include <array>

int main()
{
    std::array<int, 3> array = {{0, 1, 2}};
    auto iter = array.cbegin();
    return *iter;
}
```

clang-tidy output:
```
[<source>:6:5: warning: 'auto iter' can be declared as 'const auto *iter' [readability-qualified-auto]]
```

Godbolt demo: https://godbolt.org/z/Taraa7zxK

**Comment by PiotrZSL** [comment]
Yes, would be good to ignore iterators. Or even better have configurable list of ignored types.

**Comment by sfc-gh-sgiesecke** [comment]
The same problem exists with `std::string_view`

**Comment by xim** [comment]
+1, I also want something along the lines `readability-qualified-auto.IgnoreTypes: 'iterator'`
